torch/csrc/jit/operator.cpp

```diff
@@ -22,11 +22,22 @@ struct SchemaParser {
     std::vector<Argument> returns;
     std::vector<Symbol> writes;
     bool kwarg_only = false;
+    bool is_vararg = false;
     size_t idx = 0;
     parseList('(', ',', ')', [&] {
-      if(L.nextIf('*')) {
-        kwarg_only = true;
+      if (L.nextIf('*')) {
+        auto tok = L.cur();
+        if (tok.kind == TK_IDENT) {
+          is_vararg = true;
+          arguments.push_back(parseArgument(
+              idx++, /*is_return=*/false, /*kwarg_only=*/kwarg_only, writes));
+        } else {
+          kwarg_only = true;
+        }
       } else {
+        if (is_vararg) {
```

Comment:
if you do the above thing, this check would be handled by the `FunctionSchema` invariant.